pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/AuthenticationToken.java

```diff
@@ -98,4 +100,23 @@ public void start() throws PulsarClientException {
         // noop
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof AuthenticationToken)) {
+            return false;
+        }
+        AuthenticationToken that = (AuthenticationToken) o;
+        if (tokenSupplier == that.tokenSupplier){
+            return true;
+        }
+        return Objects.equals(tokenSupplier.get(), that.tokenSupplier.get());
```

Comment:
This does not look correct. The same supplier is allowed to give each time a different token.

Reply:
I don't think it's good that this class is designed to make it impossible to compare. Do you have any suggestions?

Reply:
I think in most of these authentication classes the equality does not make sense indeed. The classes are meant to be dynamically create and can contain complex state, (eg: opening connections to other systems).

Reply:
Your suggestion is correct, I will modify the content here.